pkg/tar/tar_test.go

```diff
@@ -185,23 +187,60 @@ func TestCreateTar(t *testing.T) {
 	verifyTarFile(t, tarFile, testFiles, testLinks)
 }
 
-func createTestTar(files []fileDesc, writer io.Writer) {
+func createTestTar(files []fileDesc, writer io.Writer, t *testing.T) {
```

Comment:
Since `createTestTar` can "fail", the common idiom in Go is to return an error.
So I suggest:

```diff
-func createTestTar(files []fileDesc, writer io.Writer, t *testing.T)
+func createTestTar(files []fileDesc, writer io.Writer) error
```

Then the caller handles the error.